assembl/models/idea_msg_columns.py

```diff
@@ -129,5 +124,52 @@ def ensure_ordering(cls):
         for idea in ideas:
             cls.ensure_ordering_for_idea(idea)
 
+    def get_column_synthesis(self):
+        synthesis = self.db.query(ColumnSynthesisPost
+            ).join(ColumnSynthesisPost.idea_links_of_content
+            ).filter(
+                IdeaContentLink.idea_id == self.idea_id,
+                ColumnSynthesisPost.tombstone_date == None,
+                Content.message_classifier == self.message_classifier
+            ).all()
+        return synthesis[0] if synthesis else None
+
+    def set_column_synthesis(self, subject=None, body=None, creator_id=None):
+        synthesis = self.get_column_synthesis()
+        if synthesis is None:
+            user_id = creator_id or get_current_request().authenticated_userid
+            synthesis = ColumnSynthesisPost(
+                message_classifier=self.message_classifier,
+                discussion_id=self.idea.discussion_id
+            )
+            self.db.add(synthesis)
+            idea_post_link = IdeaRelatedPostLink(
```

Comment:
Ooh, I'm not so sure about this. SynthesisPosts don't have `IdeaContentLinks`. Adding this will also cause the synthesis post of a column to be counted as a post (should it?).

Reply:
@maparent tagging you because this one needs attention. We are replaceing the langstring `header_id` in idea message columns with a foreign key to a post, which will be have a subject, body, sentiments, etc. But it is a representation of a Synthesis. Should this post have an idea content link, or should it behave like a `SynthesisPost` and avoid having links?

Reply:
The new ColumnSynthesisPost inherits from Content, not Post, to avoid to be counted and be retrieved as a top post. IdeaRelatedPostLink inherits from IdeaContentPositiveLink, this can be used on a Content object.